analyzers/tests/SonarAnalyzer.UnitTest/Helpers/MethodParameterLookupTest.cs

```diff
@@ -25,287 +25,374 @@
+        invalidOperationEx.Message.Should().Be("Sequence contains more than one element");
 
-            public abstract TInvocationSyntax[] FindInvocationsIn(string name);
-            public abstract object ExtractArgumentValue(TArgumentSyntax argumentSyntax);
-            public abstract TArgumentSyntax[] GetArguments(TInvocationSyntax invocation);
-            public abstract MethodParameterLookupBase<TArgumentSyntax> CreateLookup(TInvocationSyntax invocation, IMethodSymbol method);
+        var argumentEx = Assert.ThrowsException<ArgumentException>(() =>
+            lookupThrow.TryGetSymbol(VBCodeAnalysis.SyntaxFactory.StringLiteralExpression(VBCodeAnalysis.SyntaxFactory.StringLiteralToken(string.Empty, string.Empty)), out var parameter));
+        argumentEx.Message.Should().StartWith("argument must be of type Microsoft.CodeAnalysis.VisualBasic.Syntax.ArgumentSyntax");
+    }
 
-            protected InspectionBase(string source, AnalyzerLanguage language)
+    [TestMethod]
+    public void TestMethodParameterLookup_CS_MultipleCandidates()
+    {
+        var source = """
+            namespace Test
             {
-                Compiler = new SnippetCompiler(source, false, language);
-                MainInvocations = FindInvocationsIn("Main");
-            }
+                class TestClass
+                {
+                    void Main()
+                    {
+                        dynamic d = 42;
+                        AmbiguousCall(d);
+                    }
+                
+                    void AmbiguousCall(int p) { }
 
-            public MethodParameterLookupBase<TArgumentSyntax> CreateLookup(int invocationIndex, string expectedMethod)
-            {
-                var invocation = MainInvocations[invocationIndex];
-                var method = Compiler.SemanticModel.GetSymbolInfo(invocation).Symbol as IMethodSymbol;
-                method.Name.Should().Be(expectedMethod);
-                return CreateLookup(invocation, method);
+                    void AmbiguousCall(string p) { }
+                }
             }
+            """;
+        var compiler = new SnippetCompiler(source, false, AnalyzerLanguage.CSharp);
+        var lookup = new CSharpMethodParameterLookup(compiler.GetNodes<CSharpSyntax.InvocationExpressionSyntax>().Single(), compiler.SemanticModel);
 
-            public void CheckExpectedParameterMappings(int invocationIndex, string expectedMethod, object expectedArguments)
-            {
-                var lookup = CreateLookup(invocationIndex, expectedMethod);
-                InspectTryGetSyntax(lookup, expectedArguments, lookup.MethodSymbol);
-                InspectTryGetSymbol(lookup, expectedArguments, GetArguments(MainInvocations[invocationIndex]));
-            }
+        lookup.TryGetSyntax("p", out var expressions).Should().BeTrue();
+        expressions.Should().BeEquivalentTo(new { Identifier = new { ValueText = "d" } });
+    }
```

Comment:
We also need a test, where the parameters of the ambiguous methods are not in the same order:
```cs
void AmbiguousCall(bool a, bool b) { }
void AmbiguousCall(char b, char a) { }
```